### PR TITLE
Learningpaths UI

### PR DESCRIPTION
**Pull request by ChristopherChudzicki**
# What are the relevant tickets?
#61 

# Description (What does it do?)
This PR converts the infinite-corridor "stafflists" frontend UI to use the new `/learningpaths` API.

Some notes:
- UI Change: All learningpaths are sortable. (No distinction between "paths" and "lists". If authors don't want to sort the list, they don't have to.)
- I deleted the old stafflists / userlist code, despite the fact that this PR does not convert userlists to the new API. Since we're no longer treating userlists as learning resources, the frontends of learningpaths / userlists may diverge more. Also, since the old code is easily available in the open-discussions repo, I don't see much advantage to keeping it around here with deprecated course-catalog APIs.
- Learningpaths do not yet have an associated image

# Screenshots (if appropriate):

AddToListDialog
details_page
edit_list
listing
sorting

# How can this be tested?
Viewing the site as an admin (or anyone with `learning_path_editor` permissions):

1. Create a few learning paths at http://localhost:8063/learningpaths
    - Try various options: public, private, etc.
    - Try saving without required fields. You should get an error.
2. Adding/removing resources to/from your learning paths:
    - The only courses you can add/remove via UI are the ones on homepage; use the `+` button on resource cards.
    - You can add more resources via POST requests at http://localhost:8063/api/v1/learningpaths/PATH_ID_HERE/resources/
    - Once added, you can remove via `+` button dialog, too, either on the homepage or on the details page (see below).
3. View http://localhost:8063/learningpaths/PATH_ID_HERE/resources/ to view the resources added to your learning paths.
    - you can add/remove via `+` button
    - Try sorting the list items ("Reorder")

**Notes:** None of the listings above are paginated as of now

**As an anonymous user:** Or at least someone without `learning_path_editor`:

1. Check that  http://localhost:8063/learningpaths/ only lists public paths and has no "Create" button. 
2. Check that the "details page" (http://localhost:8063/learningpaths/PATH_ID_HERE/resources/) for public paths are viewable, but not viewable for private paths.
    - details page should not have Reorder / Edit buttons

### CODE REVIEW

**Comment by ChristopherChudzicki** [comment]
@Ferdi fyi; see screenshots. As discussed, all learning paths are sortable now.

File: frontends/mit-open/src/components/LearningResourceCard.test.tsx [hunk 1]
```diff
@@ -203,8 +70,49 @@ describe("LearningResourceCard (new interface)", () => {
...
+    {
+      userSettings: { is_learning_path_editor: false },
+      expectButton: false,
+    },
+    {
+      userSettings: { is_learning_path_editor: true },
+      expectButton: true,
+    },
+  ])(
+    "Shows LearningPaths button if and only if user has editing priveleges",
```
Comment (mbertrand) [inline frontends/mit-open/src/components/LearningResourceCard.test.tsx:92]: typo: privileges

**Review by mbertrand — APPROVED** [review]
LGTM,  just noticed a minor typo.  One highly subjective comment - it seemed a little odd to me that "Public" and "Private " have borders around them, that made me click on them just to see what would happen (selected? (Yes) editable? (No)).

**Comment by ChristopherChudzicki** [comment]
> LGTM, just noticed a minor typo. One highly subjective comment - it seemed a little odd to me that "Public" and "Private " have borders around them, that made me click on them just to see what would happen (selected? (Yes) editable? (No)).

I've removed the borders to avoid potential confusion with editable text fields. If we decide we want them, they are easy to add back in.

**Review by mbertrand — APPROVED** [review]
👍

**Pull request merged**